pkg/ebpf/tracee.go

```diff
@@ -363,10 +358,34 @@ func (t *Tracee) Init(ctx gocontext.Context) error {
 
 	// Initialize Process Tree (if enabled)
 
-	if t.config.ProcTree.Enabled {
+	if t.config.ProcTree.Source != proctree.SourceNone {
 		err = capabilities.GetInstance().Specific(
 			func() error {
 				t.processTree, err = proctree.NewProcessTree(ctx, t.config.ProcTree)
+
+				// Submit regular sched_process_{fork,exec,exit} events so process tree can be built
+				pipeEvts := func() {
+					t.eventsState[events.SchedProcessFork] = policy.AlwaysSubmit
+					t.eventsState[events.SchedProcessExec] = policy.AlwaysSubmit
+					t.eventsState[events.SchedProcessExit] = policy.AlwaysSubmit
+				}
+				// Submit signal sched_process_{fork,exec,exit} events so process tree can be built
+				signalEvts := func() {
+					t.eventsState[events.SignalSchedProcessFork] = policy.AlwaysSubmit
+					t.eventsState[events.SignalSchedProcessExec] = policy.AlwaysSubmit
+					t.eventsState[events.SignalSchedProcessExit] = policy.AlwaysSubmit
+				}
```

Comment:
What's the reason to have both? Only for analyze mode?
It feels redundant to process the same data twice

Reply:
Maximize chances of having the proctree entry whenever a pipeline event happens and needs the proctree. 

If, by any chance, the signal event hasn't happened yet, the proctree entry will be created by the fork/exec/exit pipeline event (before the entry is needed by a subsequent event).

Reply:
BTW there was a bug here (the sched_process_fork,exec,exit events would not be printed because this was too changing t.eventsState too late). I moved this logic to sooner step and it works alright.